pkg/util/cache/cache.go

```diff
@@ -38,6 +48,8 @@ type Cache[K comparable, V any] interface {
 	Contain(key K) bool
 	Set(key K, value V)
 	Remove(key K)
+	TryPin(keys ...K) ([]*cacheItem[K, V], bool)
```

Comment:
> nit: since `TryPin` would do an atomic operation, consider to name it just `Pin` here.
> 
> According to my understanding, `TryPin` is protected by a lock, but the `Unpin` from `cacheItem` is not. Is there any race-condition risk?
> 
> It would be easier to understand if we have symmetric `Pin` and `Unpin` primitives here.

`cacheItem`'s `pinCount` is protected by `atomic.Int32`, so I think there is not a race-condition risk.